NServiceBus.AzureStoragePersistence/issue-74/IndexPruner/Program.cs

```diff
@@ -23,39 +23,60 @@ private static CloudTable GetTable(string connectionString, string sagaTypeName)
+            Console.ReadKey();
         }
 
         public static async Task MainAsync(string[] args)
         {
-            var sagaTypes = FindAllSagaTypes();
-            if (sagaTypes.Length == 0)
+            var sagaDataTypes = FindAllSagaDataTypes();
+            if (sagaDataTypes.Length == 0)
             {
                 Console.WriteLine(
-                    "No Saga types found! Did you put all the dlls with sagas into the same directory this executable resides?");
+                    "No Saga Data Types found! Did you put all the dlls with sagas into the same directory this executable resides?");
                 return;
             }
 
-            var connectionString = ConfigurationManager.ConnectionStrings[ConnectionStringName]?.ConnectionString ??
-                                   FetchConnectionString(args);
+            var connectionString = GetConnectionString(args);
             if (string.IsNullOrWhiteSpace(connectionString))
             {
                 Console.WriteLine(
-                    $"Provide a connection string in the standard 'connectionStrings' App.config section with following name: '{ConnectionStringName}'");
+                    $"Provide a connection string in the standard 'connectionStrings' App.config section with following name: '{ConnectionStringName}' \r\nAlternatively, provide a connection string as a commandline parameter when running this tool.");
                 return;
             }
 
-            foreach (var sagaType in sagaTypes)
+            foreach (var sagaDataType in sagaDataTypes)
             {
-                Console.WriteLine($"Prunning indexes of the following saga: {sagaType}");
-                var table = GetTable(connectionString, sagaType.Name);
+                if (sagaDataType.IsGenericType)
+                {
+                    Console.WriteLine($"The saga data type `{sagaDataType.FullName}` is a generic class type and cannot be processed");
+                    continue;
+                }
+
+                Console.WriteLine($"Pruning indexes of the following saga data type: {sagaDataType}");
+                var table = GetTable(connectionString, sagaDataType.Name);
 
                 TableContinuationToken token = null;
                 do
                 {
-                    var segment = await table
-                        .ExecuteQuerySegmentedAsync(new TableQuery<SecondaryIndexTableEntity>(), token)
-                        .ConfigureAwait(false);
+                    TableQuerySegment<SecondaryIndexTableEntity> segment;
+                    try
+                    {
+                        segment = await table
+                            .ExecuteQuerySegmentedAsync(new TableQuery<SecondaryIndexTableEntity>(), token)
+                            .ConfigureAwait(false);
+                    }
+                    catch (StorageException e)
+                    {
+                        if (e.RequestInformation.HttpStatusCode == 404)
```

Comment:
Just a thought. Should we consider using `TableErrorCodeStrings.TableNotFound` to check `e.RequestInformation` whether it's truly a non existing table? [TableErrorCodeStrings](https://msdn.microsoft.com/en-us/library/microsoft.windowsazure.storage.table.protocol.tableerrorcodestrings.tablenotfound.aspx)